feat(mls): handle unsupported protocol case [WPB-5047]

vitorhugods opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5047" title="WPB-5047" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-5047</a>  [Android] Make 1-1 conversation readonly when there's no common protocol
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When a conversation is using a protocol that the device doesn't support. We need to lock the conversation UI and inform the user the reason.

### Solutions

The `MessageComposer` was been updated to handle an UnsupportedProtocol case. This case renders a disabled message composer with a warning message. This warning informs users that the conversation is using an unsupported encryption protocol.

### Dependencies

Needs releases with:

- https://github.com/wireapp/kalium/pull/2362

----
#### PR Post Merge Checklist for internal contributors

- [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

BradleyWire commented on app/src/main/scala/com/waz/zclient/LaunchActivity.scala:
```diff
@@ -96,7 +96,7 @@ class LaunchActivity extends AppCompatActivity with ActivityHelper with DerivedL
 
   private def startSignUp() = {
-    startActivity(AuthenticationActivity.newIntent(this))
+    startActivity(AuthenticationActivity.newIntent(this,false))
```
formatting

BradleyWire commented on app/src/main/scala/com/waz/zclient/preferences/pages/ProfileView.scala:
```diff
@@ -99,8 +99,7 @@ class ProfileViewImpl(context: Context, attrs: AttributeSet, style: Int) extends
       // We want to go directly to the landing page.
-      val intent = new Intent(getContext, classOf[AuthenticationActivity])
-      getContext.startActivity(intent)
+      getContext.startActivity(AuthenticationActivity.newIntent(getContext,true))
```
formatting here too

BradleyWire commented on app/src/main/res-v2/layout/activity_authentication.xml:
```diff
@@ -17,11 +17,36 @@
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <FrameLayout
+        android:id="@+id/layout_container"
```
I think we can name this to something better

mejdoo replied:
what do you suggest?

BradleyWire replied:
welcomeFragmentFrameLayoutContainer?

mejdoo replied:
too long. I guess authenticationFrameLayoutContainer is fine

gizemb commented on app/src/main/res-v2/layout/activity_authentication.xml:
```diff
@@ -17,11 +17,36 @@
+        android:layout_marginEnd="@dimen/authentication_close_button_margin"
+        android:fontFamily="@font/wire_glyphs"
+        android:gravity="center"
+        android:text="@string/glyph__close"
```
why are we still using glyphs? isn't there an svg file in zeplin for this icon?

mejdoo replied:
there is no icon in the android spec in Zeplin, just found that in the flow

gizemb replied:
https://app.zeplin.io/styleguide/5de4f201d49c777852802121/components?coid=5de4f272fca2e1782d747534 here you can find

mejdoo replied:
thanks

BradleyWire commented on app/src/main/kotlin/com/waz/zclient/auth/AuthenticationActivity.kt:
```diff
@@ -5,17 +5,37 @@ import android.content.Intent
+            Intent(context, AuthenticationActivity::class.java)
+                .putExtra(CLOSE_BUTTON_VISIBILITY_BUNDLE_KEY, isCloseButtonVisible)
+
+        private const val CLOSE_BUTTON_VISIBILITY_BUNDLE_KEY = "isCloseButtonVisible"
```
Should constants live at the bottom of a companion object?

mejdoo replied:
fixed thanks

mejdoo commented:
We have done some refactoring and this PR is no longer valid.

vitorhugods committed:
feat(mls): handle unsupported protocol case

The MessageComposer has been updated to handle an UnsupportedProtocol case. This case renders a disabled message composer with a warning message. This warning informs users that the conversation is using an unsupported encryption protocol.

AndroidBob commented:
Build [2520](https://10.10.124.134/job/AR-build-pipeline/2520/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8602353)(8602353)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8602353)(8602353)-staging-compat.apk)

vitorhugods committed:
chore: update kalium reference

AndroidBob commented:
Build [2533](https://10.10.124.134/job/AR-build-pipeline/2533/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8609517)(8609517)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8609517)(8609517)-staging-compat.apk)
